cds_ils/migrator/ill/api.py

```diff
@@ -0,0 +1,196 @@
+        raise BorrowingRequestError(
+            "found more than one library with legacy id {}".format(legacy_id)
+        )
+    else:
+        return Library.get_record_by_pid(result.hits[0].pid)
+
+
+def get_type(record):
+    """Get ILL type from legacy request_type."""
+    ill_type = "MIGRATED_UNKNOWN"
+
+    if record["request_type"] == "article":
+        ill_type = "ELECTRONIC"
+
+    if record["request_type"] == "book":
+        ill_type = "PHYSICAL_COPY"
+
+    return ill_type
+
+
+def migrate_to_ils(record):
+    """Create a record for ILS."""
+    document_pid = None
+    try:
+        item = get_item_by_barcode(record["barcode"])
+        document_pid = item.get("document_pid")
+    except ItemMigrationError:
+        document_pid = get_migration_document_pid()
+
+    # library_pid
+    library = get_library_by_legacy_id(record["id_crcLIBRARY"])
+    library_pid = library.pid.pid_value
+
+    new_record = dict(
+        document_pid=document_pid,
+        legacy_id=record.get("legacy_id"),
+        library_pid=library_pid,
+        patron_pid=get_patron_pid(record),
+        status=get_status(record),
+        type=get_type(record),
+    )
+
+    # Optional fields
+    expected_delivery_date = record.get("expected_date")
+    if expected_delivery_date:
+        new_record.update(
+            expected_delivery_date=expected_delivery_date.split("T")[0]
+        )
+
+    received_date = record.get("arrival_date")
+    if received_date:
+        new_record.update(received_date=received_date.split("T")[0])
+
+    request_date = record.get("request_date")
+    if request_date:
+        new_record.update(request_date=request_date.split("T")[0])
+
+    due_date = record.get("due_date")
+    if due_date:
+        new_record.update(due_date=due_date.split("T")[0])
+
+    total = get_cost(record)
+    if total:
+        new_record.update(total=total)
+
+    notes = get_acq_ill_notes(record)
+    if notes:
+        new_record.update(notes=notes)
+
+    budget_code = record.get("budget_code")
+    if budget_code:
+        new_record.update(budget_code=budget_code)
+
+    return new_record
+
+
+def import_ill_borrowing_requests_from_json(dump_file, include=None):
+    """Imports borrowing requests from JSON data files."""
+    dump_file = dump_file[0]
+    model, provider = model_provider_by_rectype("borrowing-request")
+    include_ids = None if include is None else include.split(",")
+
+    click.echo("Importing borrowing requests ..")
+    with click.progressbar(json.load(dump_file)) as input_data:
+        ils_records = []
+        for record in input_data:
+            if not (include_ids is None or record["legacy_id"] in include_ids):
+                continue
+
+            ils_record = import_record(
+                migrate_to_ils(record),
+                model,
+                provider,
+                legacy_id_key="legacy_id",
+            )
+            ils_records.append(ils_record)
+            ils_record.commit()
```

Comment:
do you need this extra `record.commit()`? It might not needed and it will create a second commit history. To be tested.